context/Web3Context.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { ModalId, useModalContext } from "@/context/ModalContext";
+import { PropsWithChildren, createContext, useContext, useState } from "react";
+import { waitForTransactionReceipt } from "@wagmi/core";
+import { type WriteContractParameters, type Config } from "@wagmi/core";
+
+import { writeContract as wagmiWriteContract } from "wagmi/actions";
+import { TransactionModal } from "@/app/components";
+
+export interface Web3ContextProps {
+  writeContractShowModal: (
+    config: Config,
+    args: WriteContractParameters
+  ) => Promise<string | undefined>;
+}
+
+export const Web3Context = createContext<Web3ContextProps>({
+  writeContractShowModal: async (config, args) => "",
+});
+
+export const Web3Provider = ({ children }: PropsWithChildren) => {
+  const [txHash, setTxHash] = useState<string>("");
+  const [isTxLoading, setIsTxLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const { openModal } = useModalContext();
+
+  const writeContractShowModal = async (
```

Comment:
I don't think this is a good name. I would call it, `submitTransaction`, only `writeContract`, or `startTransactionFlow`. Something around that.